apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.DetachedSessionPoolTests.cs

```diff
@@ -59,6 +59,32 @@ public void ReleaseDetachedSession_Delete()
                 // Equivalent of verify.
                 mock.Received().DeleteSessionAsync(new DeleteSessionRequest { SessionName = s_sampleSessionName }, null);
             }
+
+            [Fact]
+            public void ReleaseDetachedSession_RollbackAndDelete()
+            {
+                var logger = new InMemoryLogger();
+                var mock = SpannerClientHelpers.CreateMockClient(logger);
+                // We will force the transaction to be rollbacked and the session to be deleted, so check it happens in the mock.
+                mock.Configure()
+                    .RollbackAsync(new RollbackRequest { SessionAsSessionName = s_sampleSessionName, TransactionId = s_sampleTransactionId })
+                    .Returns(Task.CompletedTask);
+                mock.Configure()
+                    .DeleteSessionAsync(new DeleteSessionRequest { SessionName = s_sampleSessionName }, null)
+                    .Returns(Task.CompletedTask);
+
+                var pool = new SessionPool(mock, new SessionPoolOptions());
+                var session = pool.CreateDetachedSession(s_sampleSessionName, s_sampleTransactionId, ModeOneofCase.ReadWrite);
+
+                // Logically, the deletion happens asynchronously. However, everything completes synchronously so we don't need
+                // to sleep or anything to wait for the call to come through.
+                session.ReleaseToPool(true);
+                logger.AssertNoWarningsOrErrors();
+                // Equivalent of verify.
+                mock.Received().RollbackAsync(new RollbackRequest { SessionAsSessionName = s_sampleSessionName, TransactionId = s_sampleTransactionId });
+                mock.Received().DeleteSessionAsync(new DeleteSessionRequest { SessionName = s_sampleSessionName }, null);
+                
```

Comment:
Nit: trailing blank line. Happy to let this go though, if it's the only change you'd need to make.

Reply:
Done